File: iterator/c-sharp/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Iterator
{
    // Liste üzerinde gezilecek olan elamnı temsil eden sınıf.
    class Employee
    {
        public int id;
        public string firstName;
        public string lastName;

        public Employee(int id, string firstName, string lastName)
        {
            this.id = id;
            this.firstName = firstName;
            this.lastName = lastName;
        }
    }
}

File: iterator/go/employee.go
package main

// Employee Liste uzerinde gezilecek olan elemani temsil eden sinif.
type Employee struct {
	ID        int
	FirstName string
	LastName  string
}

func NewEmployee(id int, firstName string, lastName string) *Employee {
	return &Employee{
		ID:        id,
		FirstName: firstName,
		LastName:  lastName,
	}
}
